### PR TITLE
PROD-784 Updating learndash group changes privacy settings on social group

### LINKED ISSUE #2618: Updating learndash group alters privacy settings on buddyboss group
**Describe the bug**
Adding or removing users in a LearnDash group, the associated hidden group becomes a private group.

**To Reproduce**
1. Make sure that Social Group Sync and LearnDash Group Sync are enabled. Please see https://i.ibb.co/4VcQzhs/p1.png.
2. Create a Hidden Social Group
3. Create a LearnDash Group and associate it to the Social group you just created. https://i.ibb.co/HKsY8cD/p2.png
4. Try to add some users to the LearnDash group. https://i.ibb.co/7KRKhvG/p3.png
5. Now go to the Social Group and notice that it is now Private instead of Hidden.

Able to replicate it on local and dev sites.

**Note:**
It works fine when the Social Group Sync is disabled. For reference, please see https://i.ibb.co/bQVq3Gx/p5.png

**Screenshots**
https://drive.google.com/file/d/1VvnPHSs1CUjo-ITTwHbxkEAn4ShA5NyL/view?usp=sharing

**Support ticket links**
https://secure.helpscout.net/conversation/1570672169/153215/

**Jira issue** : [PROD-784]

[PROD-784]: https://buddyboss.atlassian.net/browse/PROD-784?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by mahedihasannoman**
### All Submissions:
https://buddyboss.atlassian.net/browse/PROD-4905

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2618  .

### How to test the changes in this Pull Request:

1. Make sure that Social Group Sync and LearnDash Group Sync are enabled. Please see https://i.ibb.co/4VcQzhs/p1.png.
2. Create a Hidden Social Group
3. Create a LearnDash Group and associate it to the Social group you just created. https://i.ibb.co/HKsY8cD/p2.png
4. Try to add some users to the LearnDash group. https://i.ibb.co/7KRKhvG/p3.png
5. Now go to the Social Group and notice that it is now Hidden as it was.

### Proof Screenshots or Video

https://www.loom.com/share/a8a3b1dc05d549b3bc3532280d670f97

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-integrations/learndash/library/SyncGenerator.php [hunk 1]
```diff
@@ -845,7 +845,7 @@ public function updateBuddypressGroup( $ld_group_id, $groupId ) {
 					'group_id'    => $groupId,
 					'creator_id'  => $ldGroup->post_author,
 					'name'        => $ldGroup->post_title ?: "For Social Group: {$this->ldGroupId}",
-					'status'      => $settings->get( 'learndash.default_bp_privacy' ),
+					//'status'      => $settings->get( 'learndash.default_bp_privacy' ),
```
Comment (dipskakadiya) [inline src/bp-integrations/learndash/library/SyncGenerator.php:848]: @mahedihasannoman I don't understand why we remove status sync.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-integrations/learndash/library/SyncGenerator.php (hunk 1 above)
Comment (mahedihasannoman) [reply src/bp-integrations/learndash/library/SyncGenerator.php:848]: Hi @dipskakadiya , It is not syncing status from LD group, It is adding the default status from sync settings for every LD group update. See the screenshot: https://nimb.ws/LiCa2l
and this is the issue that have been reported here.

Also according to the settings label (marked on the screenshot) this status should be assigned upon group creation only.

**Review by dipskakadiya — APPROVED** [review]

**Comment by aprilgeller** [comment]
Retested on QA ongoing branch. 
Status: Not yet Fixed

Proof Video recording: https://watch.screencastify.com/v/prQXIuPaOIEFvTg9xLHa

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
See here https://github.com/buddyboss/buddyboss-platform/pull/2967#issuecomment-971355627

**Comment by mahedihasannoman** [comment]
Hi @aprilgeller It is working fine on my testing. I have tested on qa-ongoing branch. Here is the video: https://www.loom.com/share/a0f2e3c605b943e9b856b3adc2f45f97

CC: @chetansatasiya

File: src/bp-integrations/learndash/library/SyncGenerator.php [hunk 2]
```diff
@@ -859,7 +859,7 @@ public function updateBuddypressGroup( $ld_group_id, $groupId ) {
 					'group_id'    => $groupId,
 					'creator_id'  => $ldGroup->post_author,
 					'name'        => $ldGroup->post_title ?: "For Social Group: {$this->ldGroupId}",
```
Comment (chetansatasiya) [inline src/bp-integrations/learndash/library/SyncGenerator.php:861]: @mahedihasannoman can you make this **For Social Group** string translatable please?

**Review by KartikSuthar — APPROVED** [review]

**Pull request merged**